src/Apps/Order/Routes/Review/index.tsx

```diff
@@ -113,48 +120,53 @@ export class ReviewRoute extends Component<ReviewProps, ReviewState> {
-              if (error) {
-                switch (error.code) {
-                  case "insufficient_inventory": {
-                    const artistId = this.artistId()
-                    this.onMutationError(
-                      new ErrorWithMetadata(error.code, error),
-                      "Not available",
-                      "Sorry, the work is no longer available.",
-                      artistId ? this.routeToArtistPage.bind(this) : null
-                    )
-                    break
-                  }
-                  case "failed_charge_authorize": {
-                    const parsedData = JSON.parse(error.data)
-                    this.onMutationError(
-                      new ErrorWithMetadata(error.code, error),
-                      "An error occurred",
-                      parsedData.failure_message
-                    )
-                    break
-                  }
-                  case "artwork_version_mismatch": {
-                    this.onMutationError(
-                      new ErrorWithMetadata(error.code, error),
-                      "Work has been updated",
-                      "Something about the work changed since you started checkout. Please review the work before submitting your order.",
-                      this.routeToArtworkPage.bind(this)
-                    )
-                    break
-                  }
-                  default: {
-                    this.onMutationError(
-                      new ErrorWithMetadata(error.code, error)
-                    )
-                    break
+  onOfferOrderSubmitted() {
+    if (this.props.relay && this.props.relay.environment) {
+      this.setState({ isSubmitting: true }, () =>
+        commitMutation<ReviewSubmitOfferOrderMutation>(
+          this.props.relay.environment,
+          {
+            mutation: graphql`
+              mutation ReviewSubmitOfferOrderMutation(
+                $input: SubmitOrderWithOfferInput!
+              ) {
+                ecommerceSubmitOrderWithOffer(input: $input) {
+                  orderOrError {
+                    ... on OrderWithMutationSuccess {
+                      order {
+                        state
+                      }
+                    }
+                    ... on OrderWithMutationFailure {
+                      error {
+                        type
+                        code
+                        data
+                      }
+                    }
                   }
                 }
-              } else {
-                this.onSuccessfulSubmit()
               }
+            `,
+            variables: {
+              input: {
+                offerId: this.props.order.myLastOffer.id,
+              },
+            },
+            onCompleted: result => {
+              const {
+                ecommerceSubmitOrderWithOffer: { orderOrError },
+              } = result
+              this.onSubmitCompleted(orderOrError)
```

Comment:
Totally possible-- some of them are the same, and since we haven't yet fleshed out which ones we want to handle specially, I think we can wait to implement something until we do.